utils/anautilities.py

```diff
@@ -442,6 +442,59 @@ def filePathExists(searchPath, subPath=None, debug=False):
             print "Found %s"%s(testPath)
         return True
 
+#========================================================================================
+# Find Inflection Point /////////////////////////////////////////////////////////////////
+#----------------------------------------------------------------------------------------
+# Author(s): Brendan Regnery, Brian Dorney, Jared Sturdy ////////////////////////////////
+#----------------------------------------------------------------------------------------
+
+def findInflectionPts(graph):
+    '''
+    Find and return the "knee", inflection point, of the SBit Rate scan
+    graph should be a TGraph or TH1F of the SBit Rate scan for 1 vfat 
+    '''
+
+
+    import ROOT as root
+    import numpy as np
+    from itertools import groupby
+
+    # make TH1F into TGraph
+    if type(graph) == root.TH1F :
+        graph = root.TGraph(graph)
+
+    # make arrays out of the x and y components
+    x = graph.GetX()
+    y = graph.GetY()
+    x = np.array(x, dtype=float)
+    y = np.array(y, dtype=float)
+
+    # Calculate the gradient of y as a function of x
+    # Documentation here: https://docs.scipy.org/doc/numpy/reference/generated/numpy.gradient.html
+    grad = np.gradient(y, x)
+
+    # Split the array when the slope changes sign
+    posGrad = [list(g) for k, g in groupby(grad, lambda x: x > 0) if k]
+    negGrad = [list(g) for k, g in groupby(grad, lambda x: x < 0) if k]
+   
+    # Sum the split arrays and find the most negative sum
+    negSum = []
+    bigNegSum = 0
+    for iNegArr in range(0, len(negGrad) ) :
+        tmpNegSum = sum(negGrad[iNegArr] )
+        negSum.append(tmpNegSum)
+        if tmpNegSum < bigNegSum :
+            bigNegSum = tmpNegSum
+            bigIdx =  iNegArr
+
+    # Get the inflection point
+    # We are defining the inflection point as the point where the most negative gradient sum begins
+    inflxGrad = negGrad[bigIdx][0]
+    for iGrad in range(0, len(grad) ) :
+        if grad[iGrad] == inflxGrad :
+            inflxIdx = iGrad
```

Comment:
Wouldn't it be more efficient to replace lines 493-495 with [`list::index()`](https://docs.python.org/2/tutorial/datastructures.html)? e.g.

```python
try:
   inflxIdx = grad.index(inflxGrad)
except ValueError
   print("Warning: inflxGrad value {:f} not found in grad container".format(inflxGrad))
   # Do appropriate error handling
```